src/raft.rs

```diff
@@ -346,15 +391,16 @@ impl<T: Storage> Raft<T> {
         let term = r.term;
         r.become_follower(term, INVALID_ID);
         info!(
```

Comment:
em, look wired for me, if possible, I prefer using something like `r.logger("format", args...)`, but seem it can't work, so maybe we can define a macro like `sinfo!()` which hides `r.logger`?

Reply:
Does it have runtime overhead? And it looks clumsy copycat of scala's string interpolation.

Reply:
@siddontang We can use something like https://docs.rs/slog-scope/4.0.1/slog_scope/, however it explicitly states:

> It is generally advised NOT to use `slog_scope` in libraries. Read more in slog-rs FAQ

It may be possible to roll our own that uses an implicit `self.logger` if we really wanted.

In general I don't see this as a bad thing though, it's not much different than the `write!()` macro and it allows us to, if we choose to, have different loggers with different values attached to them.